lesson22/js/index.js

```diff
@@ -0,0 +1,168 @@
+"use strict";
+
+const tableWrapper = document.getElementById("js-wrapper");
+const userTableState = {
+  orderState: {
+    id: "BOTH",
+    age: "BOTH",
+  },
+};
+
+const arrowImageSources = {
+  BOTH: "img/both.svg",
+  ASC: "img/asc.svg",
+  DESC: "img/desc.svg",
+};
+
+const request = async () => {
+  const resourceUrl = "https://jsondata.okiba.me/v1/json/t2T2M210511130923";
+  const resource = await fetch(resourceUrl);
+  let json = await resource.json();
+  return json;
+};
+
+const fetchData = async () => {
+  let res;
+  try {
+    res = await request();
+  } catch (error) {
+    res = [];
+  } finally {
+    return res;
+  }
+};
+
+const init = async () => {
+  const datas = await fetchData();
+  if (datas.length === 0) {
+    tableWrapper.textContent = "コンテンツがありません";
+    return;
+  }
+
+  createTable(datas);
+
+  const tbody = document.querySelector("tbody");
+  const sortButtons = document.querySelectorAll(".sortArrow");
+  attachClickEventForSortBtn(datas, sortButtons, tbody);
+};
+
+init();
+
+const createTable = (datas) => {
+  let savedTableParts = "";
+  const users = [...datas.data];
+  const table = document.createElement("table");
+
+  savedTableParts += createTableHeadContents();
+  savedTableParts += createTableBodyContents(users);
+  table.innerHTML = savedTableParts;
+
+  tableWrapper.appendChild(table);
+};
+
+const attachClickEventForSortBtn = (datas, sortButtons, tbody) => {
+  sortButtons.forEach((sortButton) => {
+    sortButton.addEventListener("click", () => {
+      const tableColumnId = sortButton.parentNode.id;
+      changeTableStateToNext(userTableState.orderState[tableColumnId]);
+      changeTableHeaderShow(changeTableStateToNext(userTableState.orderState[tableColumnId]), tableColumnId, sortButton);
+      sortTable(datas, tableColumnId);
+      changeTableBodyShow(sortTable(datas, tableColumnId), tbody);
+    });
+  });
+};
+
+const createTableHeadContents = () => {
+  let tableHeadContent = `
+  <thead>
+  <tr>
+  <th id="id">ID
+    <img class="sortArrow" src="img/both.svg">
+  </th>
+  <th id="name">名前</th>
+  <th id="gender">性別</th>
+  <th id="age">年齢
+    <img class="sortArrow" src="img/both.svg">
+  </th>
+  </tr>
+  </thead>
+  `;
+  return tableHeadContent;
+};
```

Comment:
`const` じゃないですかね。もしくはそのまま返す

```suggestion
 return `
  <thead>
  <tr>
  <th id="id">ID
    <img class="sortArrow" src="img/both.svg">
  </th>
  <th id="name">名前</th>
  <th id="gender">性別</th>
  <th id="age">年齢
    <img class="sortArrow" src="img/both.svg">
  </th>
  </tr>
  </thead>
  `;
};
```

Reply:
修正しました。 5f66809